src/main/java/frc/robot/subsystems/Arm.java

```diff
@@ -0,0 +1,159 @@
+package frc.robot.subsystems;
+
+import static edu.wpi.first.wpilibj2.command.Commands.runOnce;
+
+import java.time.Instant;
+import java.util.function.DoubleSupplier;
+
+import org.littletonrobotics.junction.Logger;
+
+import com.chopshop166.chopshoplib.commands.SmartSubsystemBase;
+import com.chopshop166.chopshoplib.motors.SmartMotorController;
+import com.fasterxml.jackson.annotation.JacksonInject.Value;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.Commands;
+import frc.robot.maps.subsystems.ArmMap;
+import frc.robot.maps.subsystems.ArmMap.Data;
+
+public class Arm extends SmartSubsystemBase {
+
+    Data data;
+    ArmMap map;
+    private SmartMotorController extendMotor = new SmartMotorController();
+    private final double SOFT_MAX_DISTANCE = 400.0;
+    private final double SOFT_MIN_DISTANCE = 20.0;
+    PIDController pid = new PIDController(0.03, 0, 0);
+
+    public Arm(ArmMap map) {
+        this.map = map;
+        this.data = new Data();
+    }
+
+    enum Level {
+        // 0 in.
+        LOW(0, 0),
+        // 34 in.
+        MEDIUM(0, 0),
+        // 41 7/8 in.
+        HIGH(0, 0);
+
+        private double length;
+        private double angle;
+
+        private Level(double length, double angle) {
+            this.length = length;
+            this.angle = angle;
+        }
+
+        public double get() {
+            return length;
+        }
+
+        public double getAngle() {
+            return angle;
+        }
+    }
+
+    public final double SPEED = 0.3;
+
+    public CommandBase open() {
+        return runOnce(() -> {
+            data.setPoint = 2;
+        });
+    }
+
+    public CommandBase extend(DoubleSupplier motorSpeed) {
+        return run(() -> {
+            data.setPoint = softLimit(motorSpeed.getAsDouble());
+        });
+    }
+
+    public CommandBase moveToDistance(double distance, double speed) {
+        return cmd("Move Distance").onInitialize(() -> {
+            if (distance >= data.distanceInches) {
+                // extend
+                data.setPoint = softLimit(pid.calculate(distance - data.distanceInches));
+            }
+        }).runsUntil(() -> Math.abs(distance - data.distanceInches) < 0.5).onEnd(() -> {
+            data.setPoint = 0;
+        });
+    }
+
+    private double softLimit(double speed) {
```

Comment:
Private functions at the bottom, please.
Also, remind us to teach you about Modifiers